src/engine/SpatialJoin.cpp

```diff
@@ -65,53 +53,35 @@ SpatialJoin::SpatialJoin(
 }
 
 // ____________________________________________________________________________
-void SpatialJoin::parseMaxDistance() {
-  const std::string& input = triple_.p_._iri;
-
-  if (ctre::match<MAX_DIST_IN_METERS_REGEX>(input)) {
-    std::string number = input.substr(
-        MAX_DIST_IN_METERS.size(),
-        input.size() - MAX_DIST_IN_METERS.size() - 1);  // -1: compensate for >
-    maxDist_ = std::stoll(number);
-  } else {
-    AD_THROW(
-        "parsing of the maximum distance for the "
-        "SpatialJoin operation was not possible");
-  }
-
-  if (maxDist_ < 0) {
-    AD_THROW("the maximum distance between two objects must be >= 0");
-  }
-}
+void SpatialJoin::parseConfigFromTriple() {
+  // Helper to convert a ctre match to an integer
+  auto matchToInt = [](std::string_view match) -> std::optional<long long> {
+    if (match.size() > 0) {
+      double res = 0;
+      absl::from_chars(match.data(), match.data() + match.size(), res);
+      return static_cast<long long>(res);
+    }
+    return std::nullopt;
+  };
 
-// ____________________________________________________________________________
-void SpatialJoin::parseNearestNeighbors() {
   const std::string& input = triple_.p_._iri;
-  if (auto match = ctre::search<NEAREST_NEIGHBORS_REGEX>(input)) {
-    double maxResults = -1;
-    double maxDist = -1;
-
-    std::string_view maxResultsMatch = match.get<"results">();
-    std::string_view maxDistMatch = match.get<"dist">();
-
-    absl::from_chars(maxResultsMatch.data(),
-                     maxResultsMatch.data() + maxResultsMatch.size(),
-                     maxResults);
-    if (maxDistMatch.size() > 0) {
-      absl::from_chars(maxDistMatch.data(),
-                       maxDistMatch.data() + maxDistMatch.size(), maxDist);
-    }
 
-    maxResults_ = static_cast<long long>(maxResults);
-    maxDist_ = static_cast<long long>(maxDist);
-  } else {
-    AD_THROW("invalid arguments for <nearest-neighbors>");
+  if (auto match = ctre::match<MAX_DIST_IN_METERS_REGEX>(input)) {
+    auto maxDist = matchToInt(match.get<"dist">());
+    if (maxDist.has_value()) {
```

Comment:
Can't this be `AD_CORRECTNESS_CHECK(maxDist.has_value)` because of the structureof the regex?